api/src/main/java/org/pmiops/workbench/cohorts/CohortMaterializationService.java

```diff
@@ -62,10 +81,75 @@ public CohortMaterializationService(BigQueryService bigQueryService,
     return participantIds;
   }
 
+  private ColumnConfig findPrimaryKey(TableConfig tableConfig) {
+    for (ColumnConfig columnConfig : tableConfig.columns) {
+      if (columnConfig.primaryKey) {
+        return columnConfig;
+      }
+    }
+    throw new IllegalStateException("Table lacks primary key!");
+  }
+
+  private TableQueryAndConfig getTableQueryAndConfig(FieldSet fieldSet) {
+    TableQuery tableQuery;
+    if (fieldSet == null) {
+      tableQuery = new TableQuery();
+      tableQuery.setTableName(PERSON_TABLE);
+      tableQuery.setColumns(ImmutableList.of(PERSON_ID));
+    } else {
+      tableQuery = fieldSet.getTableQuery();
+      if (tableQuery == null) {
+        // TODO: support other kinds of field sets besides tableQuery
+        throw new BadRequestException("tableQuery must be specified in field sets");
+      }
+      String tableName = tableQuery.getTableName();
+      if (Strings.isNullOrEmpty(tableName)) {
+        throw new BadRequestException("Table name must be specified in field sets");
+      }
+    }
+    CdrSchemaConfig cdrSchemaConfig = cdrSchemaConfigProvider.get();
+    TableConfig tableConfig = cdrSchemaConfig.cohortTables.get(tableQuery.getTableName());
```

Comment:
Would be good to return a BadRequest if we don't know about this table name. For now I think this would just look like a 500